### PR TITLE
Make the Popover scrolling and position behavior adapt to the content changes

### PR DESCRIPTION
**Pull request by youknowriad**
Requirement for #22789 

This PR updates the Popover behavior to make its height adapt to its content and refresh the max-height and position accordingly.

To test this, I added a story to the Popover component where you can increase the height of the popover dynamically.

The inserter popover (used in the canvas inserter) though was incompatible with this behavior. because we can't have both height: 100vh for the popover and have the popover adapts to height at the same time. I adapted the inserter popover styles to fix this.

I'm also pretty sure this PR fixes some dropdown/popover scrolling issues (more menu, block settings). I'll have to retrieve the related tracked issues.

**Testing instructions**

 - Make sure all popovers show properly
 - Make sure the inserter in both the canvas and the panel shows properly.

### CODE REVIEW

File: packages/components/src/popover/index.js [hunk 1]
```diff
@@ -260,11 +260,11 @@ const Popover = ( {
 	const anchorRefFallback = useRef( null );
 	const contentRef = useRef( null );
 	const containerRef = useRef();
-	const contentRect = useRef();
 	const isMobileViewport = useViewportMatch( 'medium', '<' );
 	const [ animateOrigin, setAnimateOrigin ] = useState();
 	const slot = useSlot( __unstableSlotName );
 	const isExpanded = expandOnMobile && isMobileViewport;
+	const [ containerResizeListener, contentSize ] = useResizeObserver();
```
Comment (ellatrix) [inline packages/components/src/popover/index.js:267]: I guess we can't use https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver?

Comment (youknowriad) [reply packages/components/src/popover/index.js:267]: Not for the moment (IE11) but this hook is widely used in Gutenberg already and it works pretty well.

Comment (ellatrix) [reply packages/components/src/popover/index.js:267]: We also already have an interval running, so we could use `ResizeObserver` and have the interval as a fallback.

Comment (youknowriad) [reply packages/components/src/popover/index.js:267]: Why such complexity while the hook is already there and works properly? I'd rather find a way to remove the interval personally.

Comment (ellatrix) [reply packages/components/src/popover/index.js:267]: Yes, would be great if we could remove the interval

**Review by ellatrix — APPROVED** [review]
Looks good! I thought this might undo the size adjustments to fit the popover within the viewport, but I see that they're still scrollable.

**Pull request merged**